driver/terraform.go

```diff
@@ -224,24 +225,78 @@ func (tf *Terraform) ApplyTask(ctx context.Context) error {
 }
 
 // UpdateTask updates the task on the driver. Makes any calls to re-init
-// depending on the fields updated.
-func (tf *Terraform) UpdateTask(patch PatchTask) error {
+// depending on the fields updated. If update task is requested with the inspect
+// run option, then returns the inspected plan
+func (tf *Terraform) UpdateTask(ctx context.Context, patch PatchTask) (string, error) {
+	switch patch.RunOption {
+	case "", RunOptionInspect, RunOptionNow:
+		// valid options
+	default:
+		return "", fmt.Errorf("Invalid run option '%s'. Please select a valid "+
+			"option", patch.RunOption)
+	}
+
 	tf.mu.Lock()
 	defer tf.mu.Unlock()
 
 	reinit := false
 
 	if tf.task.Enabled != patch.Enabled {
 		tf.task.Enabled = patch.Enabled
-		reinit = true
+		if tf.task.Enabled {
+			reinit = true
+		}
+	}
+
+	// identify cases where resources are not impacted and we can return early
+	switch {
+	case patch.Enabled == false:
+		return "", nil
+	default:
+		// onward!
 	}
 
 	if reinit {
+		if err := tf.initTask(true); err != nil {
+			return "", fmt.Errorf("Error updating task '%s'. Unable to init "+
+				"task: %s", tf.task.Name, err)
+		}
 		tf.inited = false
```

Comment:
This took me a couple of reads to understand `tf.initTask` is different than `tf.inited` because it's been a while since I've viewed this file. Not an ask but something in the future we can do is add context to what `inited` is pertaining to, the Terraform workspace.

Reply:
Agree, we have a lot of 'init's going on. This tf.inited is an optimization to track if we've done a `terraform init` so that we don't need to call it every time before `terraform apply`. One thing I can do now, I'll move the `tf.inited = false` within tf.initTask() since it's likely every time we call tf.initTask(), we will likely need to call `terraform init` again.

Reply:
I like that minor change, it makes sense.